### DISCUSSION TITLE
Rule proposal: buttons should have type specified

### DISCUSSION THREAD

**Discussion by bbarry** [angular-eslint/angular-eslint#823] [New Rule Proposals]
We repeatedly are finding that a developer here or there has forgotten to add a `type` attribute to `<button>` tags in templates. Doing so causes buttons to trigger form submit events when they complete and happen to have an ancestor element  that is a form tag. This may result in unintended code executing when someone presses the button. Additionally a browser or assisting device might choose to trigger the button to activate when a person submits from within a field in the form.

Valid code:

```html
    <button type="submit" ...
    <button type="button" ...
    <button type="reset" ...
```

invalid:

```html
    <button>
    <button type="">
    <button type="somethingelse">
```

**Comment by chernodub** [comment]
I've created a [PR on this](https://github.com/angular-eslint/angular-eslint/pull/928), please check
